refactor: logging using string interpolation - WPB-14876

caldrian opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-14876" title="WPB-14876" target="_blank"><img alt="Story" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10815?size=medium" />WPB-14876</a>  [iOS] Create a new WireLogger interface
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove the jira markers to link tickets automatically -->

### Issue

This is the third PR in a series,
- the previous one is: https://github.com/wireapp/wire-ios/pull/3849

This PR adjusts WireBackup to use the new logging module.

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

mikeger commented:
We need to merge this after the release 3.18 is done.

marcoconti83 commented on Wire-iOS/Sources/Helpers/syncengine/ZMUserSession+iOS.m:
```diff
@@ -24,7 +24,7 @@
 const static unsigned long long MaxFileSize = 25 * 1024 * 1024 - 32; // 25 megabytes - 32 bytes for IV and padding
```
It's not necessary to remove 32 bytes from the max size to make space for the IV. 
The encrypted file can be 25MB + 32 bytes. Otherwise the plain text max file size from the user point of view is not 25MB but 25MB - 32 bytes, which is not easy to explain.
The backend will not complain for the extra 32 bytes :)

nicorsm replied:
Thanks for the review @marcoconti83!

marcoconti83 commented on Wire-iOS/Sources/Helpers/syncengine/ZMUserSession+iOS.m:
```diff
@@ -24,7 +24,7 @@
 
 // Temporarily set to have the same size as non-team.
-const static unsigned long long MaxTeamFileSize = 25 * 1024 * 1024 - 32; // 25 megabytes - 32 bytes for IV and padding
+const static unsigned long long MaxTeamFileSize = 100 * 1024 * 1024 - 32; // 100 megabytes - 32 bytes for IV and padding
```
Same comment as above, no need to remove the 32 bytes.

caldrian committed:
Merge branch 'release/cycle-3.114' of github.com:wireapp/wire-ios into chore/move-WireLogger-into-WireFoundation-WPB-10941

caldrian committed:
brainstorming

caldrian committed:
WireAnalytics.setup()

caldrian committed:
add Sendable conformance

caldrian committed:
Merge branch 'fix/make-WireLogger-Swift6-ready-WPB-11890' into chore/abstract-logging

caldrian committed:
create targets WireLogging etc.

caldrian committed:
move file

caldrian committed:
Merge branch 'release/cycle-3.114' of github.com:wireapp/wire-ios into chore/abstract-logging

caldrian committed:
attempt to extend WireLogInterpolation from outside

samwyndham committed:
fix: Crash unwrapping `MainCoordinator.splitViewController` - WPB-14510 (#2207)

samwyndham committed:
fix: Profile color tappable area - WPB-14525 (#2208)

caldrian committed:
chore: delete redundant LICENSE and README files - WPB-14567 (#2214)

netbe committed:
fix: back button style - WPB-14478 (#2213)

netbe committed:
chore: bump avs to 9.10.16 - WPB-11904 (#2118)

KaterinaWire committed:
feat: Add scroll view to the passcode setup view - WPB-11836 (#2211)

caldrian committed:
Merge branch 'release/cycle-3.114' of github.com:wireapp/wire-ios into chore/move-WireLogger-into-WireFoundation-WPB-10941

caldrian committed:
fix: remove debug logging of credentials - WPB-14251 (#2205)

caldrian committed:
add missing imports

caldrian committed:
chore: unify schemes and test plans - WPB-11406 (#2217)

caldrian committed:
fix: crash when registering with a registered email - WPB-14566 (#2223)

caldrian committed:
Merge branch 'release/cycle-3.114' of github.com:wireapp/wire-ios into chore/abstract-logging

caldrian committed:
add notes

caldrian committed:
squashed commit

# Conflicts:
#	WireDomain/Sources/WireDomain/Repositories/UserClients/Models/UserClientInfo.swift
#	WireFoundation/Tests/WireLoggingTests/PlaceholderTests.swift
#	wire-ios-system/Source/Logging/WireLogger.swift
#	wire-ios/Wire-iOS Tests/WireAnalytics_DatadogTests.swift
#	wire-ios/Wire-iOS.xcodeproj/project.pbxproj
#	wire-ios/WireCommonComponents/Analytics/WireAnalytics.swift
#	wire-ios/WireCommonComponents/Logging/CocoaLumberjackLogger.swift

caldrian committed:
Merge commit 'b0666045b3a660df08cc667bb0b5497154303bc2' into chore/abstract-logging

caldrian committed:
fix merge conflicts

caldrian committed:
add debug output

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/abstract-logging

caldrian committed:
some progress

caldrian committed:
restructuring

caldrian committed:
StaticString

caldrian committed:
cleanup

caldrian committed:
sendable

caldrian committed:
rename WireLogger to OldWireLogger

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/abstract-logging

caldrian committed:
fix build assets reference

caldrian committed:
start over from abstract-logging branch

caldrian committed:
Merge branch 'chore/abstract-logging' into chore/move-WireLogger-into-WireFoundation-WPB-10941

caldrian committed:
wip

caldrian committed:
some progress

caldrian committed:
some more progress

caldrian committed:
OSLogLoggingSystem

caldrian committed:
cleanup

caldrian committed:
code style

caldrian committed:
prepare for refactoring

caldrian committed:
refactoring

caldrian committed:
adapters

caldrian committed:
remove commented code

caldrian committed:
cleanup

caldrian committed:
additional attributes

caldrian committed:
further cleanup

caldrian committed:
use logger in WireSystem

caldrian committed:
allow for skipping obfuscation for errors

caldrian committed:
finish some tests

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/move-WireLogger-into-WireFoundation-WPB-10941

# Conflicts:
#	WireFoundation/Tests/TestPlans/AllTests.xctestplan

caldrian committed:
improve directory structure

caldrian committed:
reduce the number of changes

caldrian committed:
undo changes

caldrian committed:
reverts

caldrian committed:
WireLogger.with(tag:)

caldrian committed:
revert changes

caldrian committed:
create target WireLogging

caldrian committed:
move several files

caldrian committed:
move SystemLogger

caldrian committed:
move all files except objc

caldrian committed:
add missing imports

caldrian committed:
fix imports in DataModel

caldrian committed:
fix imports in WireDomain

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/move-WireLogger-into-WireFoundation-WPB-10941

caldrian committed:
fix imports in RequestStrategy

caldrian committed:
fix imports in SyncEngine

caldrian committed:
add missing imports

caldrian committed:
format code

caldrian committed:
Trigger CI

caldrian committed:
rename WireLoggerObjc to WireLoggerObjC

caldrian committed:
fix filename

caldrian committed:
move file to avoid issues

caldrian committed:
convert groups to folders

caldrian committed:
undo renaming

caldrian committed:
Merge branch 'chore/move-WireLogger-into-WireFoundation-WPB-10941' into chore/create-WireLogging-target-WPB-14297

# Conflicts:
#	WireFoundation/Sources/WireLogging/WireLogger.swift
#	WireFoundation/Sources/WireLoggingSupport/WireDomainPackage.swift
#	wire-ios-sync-engine/Source/SessionManager/APIMigration/APIMigrationManager.swift
#	wire-ios-sync-engine/Source/SessionManager/APIMigration/AccessTokenMigration.swift
#	wire-ios-sync-engine/WireSyncEngine.xcodeproj/project.pbxproj
#	wire-ios-system/WireSystem.xcodeproj/project.pbxproj
#	wire-ios-transport/Source/TransportSession/UnauthenticatedTransportSession.swift
#	wire-ios/Wire-iOS.xcodeproj/project.pbxproj
#	wire-ios/Wire-iOS/Sources/UserInterface/Calling/CallGridView/CallParticipantViews/AVSVideoContainerView.swift

caldrian committed:
move existing files into subdirectory

caldrian committed:
add Package.resolved to .gitignore

caldrian committed:
link WireLogging to unit tests

caldrian committed:
add TODO, clean up (null) in xcodeproj

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/move-WireLogger-into-WireFoundation-WPB-10941

caldrian committed:
create package WireLogging

caldrian committed:
WireLoggingAll scheme

caldrian committed:
fix framework.rb

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/move-WireLogger-into-WireFoundation-WPB-10941

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/move-WireLogger-into-WireFoundation-WPB-10941

caldrian committed:
fix missing import

caldrian committed:
revert converting groups to folders

caldrian committed:
convert groups to folders again

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/create-WireLogging-target-WPB-14297

caldrian committed:
move files

caldrian committed:
move files

caldrian committed:
Merge branch 'chore/move-WireLogger-into-WireFoundation-WPB-10941.tmp' into chore/create-WireLogging-target-WPB-14297

caldrian committed:
prefix existing types with Legacy

caldrian committed:
Revert "prefix existing types with Legacy"

This reverts commit b4146df67a2412bc83e1406dedce3b74ba4683c5.

caldrian committed:
move existing code into WireLegacyLogging

caldrian committed:
revert many renamings

caldrian committed:
revert remaining renamings

caldrian committed:
add WireLogging as dependency of WireAnalytics

caldrian committed:
use WireLogger from AnalyticsService

caldrian committed:
create WireLogLevel

caldrian committed:
fix build errors

caldrian committed:
migrate WireAnalytics to Swift 6

caldrian committed:
fix more build errors

caldrian committed:
format code

caldrian committed:
Merge commit 'b2510343bd2e19709bb23066b306afb7f8a0f7b6' into chore/create-WireLogging-target-WPB-14297

caldrian committed:
Merge branch 'chore/resolve-todos-WPB-10941' into chore/create-WireLogging-target-WPB-14297

caldrian committed:
sort folders in fastlane/framework.rb

caldrian committed:
remove unused import

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/move-WireLogger-into-WireFoundation-WPB-10941

caldrian committed:
Merge branch 'chore/move-WireLogger-into-WireFoundation-WPB-10941' into chore/resolve-todos-WPB-10941

caldrian committed:
several fixes

caldrian committed:
fix redundant values

caldrian committed:
fix filenam

caldrian committed:
Merge branch 'chore/move-WireLogger-into-WireFoundation-WPB-10941' into chore/resolve-todos-WPB-10941

caldrian committed:
fix dependency tracking

caldrian committed:
Merge remote-tracking branch 'github/gh-readonly-queue/develop/pr-1938-57ac870d7a6aa55c91ab6a4edd8cd2ed7fa5f126' into chore/resolve-todos-WPB-10941

caldrian committed:
Merge branch 'chore/resolve-todos-WPB-10941' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
remove variadic parameters

caldrian committed:
WireLoggingProvider.tag

caldrian committed:
generics

caldrian committed:
delete aggregated provider builder

caldrian committed:
minor fix

caldrian committed:
fix action reference

caldrian committed:
Merge commit 'd03ca9769515b6088f45cf81b5b29526e6739a94' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
fix conflict

caldrian committed:
delete dead code

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireAnalytics/Package.swift
#	wire-ios/Wire-iOS.xcodeproj/project.pbxproj
#	wire-ios/Wire-iOS/Sources/Components/Settings+Logging.swift

caldrian committed:
demo - tmp

caldrian committed:
Revert "demo - tmp"

This reverts commit a63cc63518e67125098b6881a53237c1d1ebc2dc.

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	wire-ios-data-model/WireDataModel.xcodeproj/project.pbxproj
#	wire-ios-sync-engine/WireSyncEngine.xcodeproj/project.pbxproj
#	wire-ios/Wire-iOS.xcodeproj/project.pbxproj
#	wire-ios/Wire-iOS/Sources/Components/Settings+Logging.swift

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	wire-ios-transport/WireTransport.xcodeproj/project.pbxproj
#	wire-ios-utilities/WireUtilities.xcodeproj/project.pbxproj

caldrian committed:
fix: Try fix text link color - WPB-17726 (#3047)

Authored-by: Dmitry Simkin <dzmitry.simkin.ext@wire.com>

caldrian committed:
feat: cross-platform backup - WPB-16658 (#2889)

samwyndham committed:
fix: unknown symbols issue - WPB-17820 (#3050)

github-actions[bot] committed:
fix: incremental sync observer crash - WPB-17644 (#3046)

Authored-by: Jullian Mercier <31648126+jullianm@users.noreply.github.com>

github-actions[bot] committed:
fix: pull self user after pulling known users - WPB-17516 (#3043)

Co-authored-by: Jullian Mercier <31648126+jullianm@users.noreply.github.com>

samwyndham committed:
fix: link `Wire-iOS` with `WireCellsAPI` - WPB-17860 (#3057)

github-actions[bot] committed:
fix: CallKit crashes when in background - WPB-17848 (#3061)

Co-authored-by: Jullian Mercier <31648126+jullianm@users.noreply.github.com>

github-actions[bot] committed:
fix: max participants 2000 formatting with locale - WPB-17344 (#3063)

Authored-by: Dmitry Simkin <dzmitry.simkin.ext@wire.com>

caldrian committed:
Merge branch 'release/cycle-3.125' of github.com:wireapp/wire-ios into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	wire-ios-notification-engine/Sources/NotificationSession.swift
#	wire-ios-sync-engine/WireSyncEngine.xcodeproj/project.pbxproj
#	wire-ios/Wire-iOS.xcodeproj/project.pbxproj

typfel committed:
feat: batch decryption - WPB-17614 (#3014)

Co-authored-by: Christoph Aldrian <christoph.aldrian@wire.com>
Co-authored-by: Dmitry Simkin <dzmitry.simkin.ext@wire.com>
Co-authored-by: Christoph Aldrian <caldrian@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Sam Wyndham <samwyndham@users.noreply.github.com>
Co-authored-by: François Benaiteau <netbe@users.noreply.github.com>

github-actions[bot] committed:
fix: messages not being sent, waiting forever for decryption to finish - WPB-17866 (#3074)

Co-authored-by: Jullian Mercier <31648126+jullianm@users.noreply.github.com>

KaterinaWire committed:
feat: add spinner - WPB-17596 (#3069)

Co-authored-by: Christoph Aldrian <caldrian@users.noreply.github.com>

github-actions[bot] committed:
fix: push notification connection accepted placeholder - WPB-17903 (#3085)

Authored-by: Jullian Mercier <31648126+jullianm@users.noreply.github.com>

caldrian committed:
chore: delete unused targets and files (WireLoggingSupport/Tests) - WPB-17865 (#3067)

caldrian committed:
Merge branch 'release/cycle-3.125' of github.com:wireapp/wire-ios into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireLogging/Package.swift

caldrian committed:
refactor: logging using string interpolation - WPB-14297 squashed

caldrian committed:
refactor: logging using string interpolation - WPB-14297 squashed

# Conflicts:
#	WireLogging/Package.swift
#	WireLogging/Sources/WireLegacyLogging/AggregatedLogger.swift
#	WireLogging/Sources/WireLegacyLogging/SystemLogger.swift
#	wire-ios-data-model/WireDataModel.xcodeproj/project.pbxproj
#	wire-ios-notification-engine/WireNotificationEngine.xcodeproj/project.pbxproj
#	wire-ios-share-engine/WireShareEngine.xcodeproj/project.pbxproj
#	wire-ios-sync-engine/WireSyncEngine.xcodeproj/project.pbxproj
#	wire-ios-system/Source/ZMSLog.swift
#	wire-ios/Wire-iOS/Sources/UserInterface/Settings/Debug Report/LogFilesProvider.swift
#	wire-ios/WireCommonComponents/Analytics/WireAnalytics.swift

caldrian committed:
Merge branch 'refactor/WireLogging-using-string-interpolation-WPB-14297' of github.com:wireapp/wire-ios into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
Merge branch 'refactor/WireLogging-using-string-interpolation-WPB-14297' of github.com:wireapp/wire-ios into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
Trigger CI

caldrian committed:
refactor: logging using string interpolation - WPB-14297 squashed

caldrian committed:
Merge commit '48315f4de14523077c7e1a0273761ef528b1dae1' into temp-ca

caldrian committed:
fix build errors in WireLegacyLogging

caldrian committed:
format code

caldrian committed:
create WireLoggingAll scheme

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireAnalytics/Sources/WireAnalytics/WireLegacyLogging.swift
#	WireDomain/Sources/WireDomain/WireLegacyLogging.swift
#	WireLogging/Package.swift
#	WireLogging/Sources/WireLegacyLogging/WireLogMessage/WireLogInterpolation+anyError.swift
#	WireLogging/Sources/WireLegacyLogging/WireLogMessage/WireLogInterpolation+isObfuscationRequired.swift
#	WireLogging/Sources/WireLegacyLogging/WireLogMessage/WireLogInterpolation.swift
#	WireLogging/Sources/WireLegacyLogging/WireLogMessage/WireLogMessage.swift
#	WireLogging/Sources/WireLegacyLogging/WireLogMessage/WireLoggerAttribute.swift
#	WireLogging/Sources/WireLegacyLogging/WireLogMessage/WireLoggerTag.swift
#	wire-ios-notification-engine/WireNotificationEngine.xcodeproj/project.pbxproj
#	wire-ios-request-strategy/WireRequestStrategy.xcodeproj/project.pbxproj
#	wire-ios-share-engine/WireShareEngine.xcodeproj/project.pbxproj
#	wire-ios-transport/Source/Logging/WireLegacyLogging.swift
#	wire-ios-utilities/Source/WireLegacyLogging.swift
#	wire-ios/Wire Notification Service Extension/WireLegacyLogging.swift
#	wire-ios/Wire-iOS Share Extension/WireLegacyLogging.swift
#	wire-ios/Wire-iOS/Sources/Helpers/WireLegacyLogging.swift
#	wire-ios/WireCommonComponents/WireLegacyLogging.swift

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	wire-ios-data-model/WireDataModel.xcodeproj/project.pbxproj
#	wire-ios-notification-engine/WireNotificationEngine.xcodeproj/project.pbxproj
#	wire-ios-share-engine/WireShareEngine.xcodeproj/project.pbxproj

caldrian committed:
fix conflict

caldrian committed:
undo changes

caldrian committed:
move files

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	wire-ios-sync-engine/WireSyncEngine.xcodeproj/project.pbxproj

caldrian committed:
Trigger CI

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
fix merge conflicts

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
minor fix

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireLogging/Sources/WireLogging/WireLogMessage/WireLoggerAttribute.swift

caldrian committed:
use sourcery

caldrian committed:
revert deleting file

caldrian committed:
fix build error

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
revert some changes

caldrian committed:
fix build errors

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
revert changes

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
import tests

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into feat/create-new-logger-WPB-14297

# Conflicts:
#	WireLogging/Sources/WireLoggingSupport/Sourcery/sourcery.yml

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireLogging/Sources/WireLogging/LoggingProviders/WireLoggingProvider.swift
#	WireLogging/Sources/WireLogging/Placeholder.swift
#	WireLogging/Sources/WireLogging/WireLogMessage/WireLoggerAttribute.swift
#	wire-ios-data-model/Source/Model/Message/ZMMessage+Backup.swift

caldrian committed:
import files

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
rename WireLogger to WireTaggedLogger

caldrian committed:
comment test

caldrian committed:
include tag in log signature

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireLogging/Sources/WireLogging/LoggingProviders/WireLoggingProvider.swift
#	WireLogging/Sources/WireLogging/TaggedLogger/WireLogger+Singletons.swift

caldrian committed:
renaming

caldrian committed:
OSLogHandler

caldrian committed:
WireLogAttribute

caldrian committed:
additional attributes

caldrian committed:
fix logging attributes in OSLog

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireLogging/Sources/WireLogging/LogMessage/WireLogInterpolation+anyError.swift
#	WireLogging/Sources/WireLogging/LogMessage/WireLogInterpolation+isObfuscationRequired.swift

caldrian committed:
minor fix

caldrian committed:
Clean up

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireLogging/Tests/WireLoggingTests/WireLogMessage/WireLogInterpolationTests.swift

caldrian committed:
fix build errors in WireBackup

caldrian committed:
AnalyticsEvent

caldrian committed:
refactoring

caldrian committed:
minor fixes

caldrian committed:
create WireLoggingAPI

caldrian committed:
Revert "create WireLoggingAPI"

This reverts commit f96e849f23217abb721ae0f45f233e90409bee30.

caldrian committed:
fix a few build errors

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into feat/create-new-logger-WPB-14297

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
fix merge conflict

caldrian committed:
fix more build errors

caldrian committed:
move wire logger instance to extension

caldrian committed:
renaming

caldrian committed:
delete unused code

caldrian committed:
fix build errors

caldrian committed:
attempt to integrate sourcery as package

caldrian committed:
Revert "attempt to integrate sourcery as package"

This reverts commit 8904d79f7dc796a0087d5016b698af4d452e045b.

caldrian committed:
update packages

caldrian committed:
Revert "update packages"

This reverts commit 19bf28471800c0725146fdb9ebcfdb9a9bcf2276.

caldrian committed:
fix build error

caldrian committed:
Merge branch 'chore/rename-wirelogging-to-wirelegacylogging-WPB-14876' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
Merge commit '91d5c8a4e42645ff6835860c046eca5068868d55' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
logger cache

caldrian committed:
don't use NSCache

caldrian committed:
refactoring

caldrian committed:
finished

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireLogging/Sources/WireLogging/LogHandler/OSLogHandler/OSLogHandler.swift

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

caldrian committed:
SafePublicLoggable

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireAnalytics/Package.swift
#	WireNetwork/Package.swift
#	WireUI/Sources/WireSettingsUI/Account/BackupImportExport/BackupImportExportBuilder.swift
#	wire-ios/Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift

caldrian committed:
revert WireNetwork changes

caldrian committed:
fix build errors in WireBackup

caldrian committed:
fix build errors

caldrian committed:
refactoring

caldrian committed:
log process id and name with system tag

caldrian committed:
refactoring

caldrian committed:
minor corrections

caldrian committed:
more cleanup

caldrian committed:
add tests

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireLogging/Sources/WireLogging/LogMessage/WireLogAttribute.swift

caldrian committed:
cleanup

caldrian committed:
minor fix

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

netbe commented on wire-ios/WireCommonComponents/Analytics/Datadog/WireDatadog+LoggerProtocol.swift:
```diff
@@ -21,7 +21,50 @@ import WireDatadog
+            if let key = LogAttributesKey(rawValue: additionalAttribute.key) {
+                attributes[key] = additionalAttribute.value
+            } else {
+                assertionFailure(additionalAttribute.key)
```
suggestion:

```suggestion
                assertionFailure("unsupported attribute: \(additionalAttribute.key)")
```

netbe commented on wire-ios/WireCommonComponents/Logging/CocoaLumberjackLogger.swift:
```diff
@@ -21,8 +21,39 @@ import Foundation
 import WireLogging
 import WireSystem
 
+// TODO: the new implementation currently just wraps the old one, rewrite!
```
question: @caldrian do you plan to do it in this PR?

netbe commented on WireBackup/Sources/WireBackup/Extensions/WireLogging/WireLogger+instances.swift:
```diff
@@ -16,6 +16,12 @@
-typealias WireLogger = WireLegacyLogging.WireLogger
+public extension WireLogger {
+
+    static let createBackup = WireTaggedLogger(tag: "create-backup", handler: logHandler)
+
+    static let importBackup = WireTaggedLogger(tag: "import-backup", handler: logHandler)
```
question: so each framework we'll get different instances?

netbe commented on WireLogging/Tests/WireLoggingTests/WireLogMessage/WireLogInterpolation+anyErrorTests.swift:
```diff
@@ -0,0 +1,70 @@
+
+@testable import WireLogging
+
+struct WireLogInterpolationAnyErrorTests {
```
praise: nice to have this tested

caldrian committed:
Merge branch 'feat/create-new-logger-WPB-14297' into refactor/WireLogging-using-string-interpolation-WPB-14297

# Conflicts:
#	WireAnalytics/Sources/WireDatadog/WireDatadog.swift
#	WireAnalytics/Sources/WireDatadog/WireFakeDatadog.swift
#	WireLogging/Sources/WireLogging/TaggedLogger/WireLogger+Singletons.swift
#	WireUI/Sources/WireSettingsUI/Account/BackupImportExport/Import/ImportBackupViewModel.swift
